alpha/apps/kaltura/lib/myPartnerUtils.class.php

```diff
@@ -344,7 +344,7 @@ public static function getHost ( $partner_id )
 	public static function getCdnHost ( $partner_id, $protocol = null, $hostType = null )
 	{
 		// in case the request came through https, force https url
-		if (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on')
+		if (infraRequestUtils::getProtocol() == 'https')
 			$protocol = 'https';
```

Comment:
After reviewing all the cases , I can say that - 
1. In most of them value of null is passed , which will be handled as http by infraRequestUtils::getProtocol()
2. In the cases where protocol is being sent, it is either http or https. In first place the argument protocol should have been enum of 2 values.
Bottom line - lets do the full change , review again all the code that sends none null value in protocol argument to getCdnHost and we will watch this case in compat check.